src/rules/rules.c

```diff
@@ -362,6 +371,7 @@ int SchemaRules_AddArgs(const char *index, const char *name, ArgsCursor *ac, Que
     SchemaRules_g->revision++;
     SchemaRules_StartScan(0);
   }
+  SchemaRules_RegisterIndex(sp);
```

Comment:
This doesn't belong here; this should be done when the index is created/loaded. I guess there's no harm in adding it here, but it should not be necessary.

Reply:
@mnunberg this is needed for ruleadd on index that did not have rules before ...

Reply:
Are we allowing this? I assumed we were still going to force users to declare an index using `WITHRULES` so that indexes added via `FT.ADD` (when 2.2 comes around, and we might still support it) won't be able to use rules and vice versa.

Reply:
Your api allows it, your tests does it.

Reply:
Hrm, that's a bug then